src/backend/functions/change-requests.ts

```diff
@@ -110,12 +111,102 @@ const getChangeRequestByID: ApiRouteFunction = async (params: { id: string }) =>
   return buildSuccessResponse(changeRequestTransformer(requestedCR));
 };
 
+const createNewChangeRequest: ApiRouteFunction = async (_params, event) => {
+  if (event.body === undefined) {
+    return buildClientFailureResponse('No data for CR creation.');
+  }
+  const body = JSON.parse(event.body!);
+  if (
+    body.submitterId === undefined ||
+    body.wbsElementId === undefined ||
+    body.type === undefined
+  ) {
+    return buildClientFailureResponse('Missing common CR field(s).');
+  }
+  if (!(body.submitterId >= 0) || !(body.wbsElementId >= 0)) {
+    return buildClientFailureResponse('ID field(s) must be integers.');
+  }
+  if (
+    body.type === CR_Type.ISSUE ||
```

Comment:
In my opinion, I think this if statement is what's most critical to understand in this code, but please correct me I'm mistaken. We want to know what type of Change Request it is and then act accordingly. If it's any of these three, we make our Prisma call and actually do stuff. If it's a STAGE_GATE or ACTIVATION, we return those respective responses without doing anything here. 

That information is harder to see with all of the validation rules inside of these if statements i.e. lines 134 - 142

Reply:
^The above is me justifying the added cost of a validation library or even taking time manually abstracting functions

Reply:
100% agreed, it's messy and tough to read this implementation